packages/clay-management-toolbar/src/ClayManagementToolbar.soy

```diff
@@ -462,7 +462,15 @@
 		viewMoreURL: string
 	]}
 
-	{if $creationMenu and (($creationMenu.primaryItems and length($creationMenu.primaryItems) == 1 and (not $creationMenu.secondaryItems or length($creationMenu.secondaryItems) == 0)) or ($creationMenu.secondaryItems and length($creationMenu.secondaryItems) == 1 and (not $creationMenu.primaryItems or length($creationMenu.primaryItems) == 0)))}
+	{let $isPrimaryItems: $creationMenu and $creationMenu.primaryItems and length($creationMenu.primaryItems) > 0 /}
+
+	{let $isSecondaryItems: $creationMenu and $creationMenu.secondaryItems and length($creationMenu.secondaryItems) > 0 /}
+
+	{let $isOnePrimaryItem: $isPrimaryItems and length($creationMenu.primaryItems) == 1 /}
+	
+	{let $isOneSecondaryItem: $isPrimaryItems and length($creationMenu.primaryItems) == 1 /}
+
+	{if ($isOnePrimaryItem and not $isSecondaryItems) or ($isOneSecondaryItem and not $isPrimaryItems)}
```

Comment:
This is a huge improvement!

From a grammatical perspective, these names might work better ("is-plural" sounds like it violates concordance, but "has-plural" does not):

- `$isPrimaryItems` -> `$hasPrimaryItems`
- `$isSecondaryItems` -> `$hasSecondaryItems`
- `$isOnePrimaryItem` -> `$hasOnePrimaryItem` 
- `$isOneSecondaryItem` -> `$hasOneSecondaryItem`

Reply:
Changed, thx @wincent !